src/rgw/rgw_rados.cc

```diff
@@ -1614,6 +1614,19 @@ void RGWPeriodMap::decode(bufferlist::iterator& bl) {
   }
 }
 
+// run an MD5 hash on the zone_id and return the first 32 bits
+static uint32_t gen_short_zone_id(const std::string zone_id)
+{
+  unsigned char md5[CEPH_CRYPTO_MD5_DIGESTSIZE];
+  MD5 hash;
+  hash.Update((const byte *)zone_id.c_str(), zone_id.size());
+  hash.Final(md5);
+
+  uint32_t short_id;
+  memcpy((char *)&short_id, md5, sizeof(short_id));
+  return std::max(short_id, 1u);
```

Comment:
@cbodley probably because I wanted to be designate 0 as a no-zone-id assigned. E.g., when an object was created on a zone that didn't have a zone id, then we'd get a 0.